processing/src/main/java/io/druid/query/aggregation/hyperloglog/HyperUniquesAggregator.java

```diff
@@ -57,7 +57,8 @@ public void reset()
   @Override
   public Object get()
   {
-    return collector;
+    // Workaround for OnheapIncrementalIndex's penchant for calling "aggregate" and "get" simultaneously.
+    return HyperLogLogCollector.makeCollector(collector.getStorageBuffer().duplicate());
```

Comment:
@drcrallen posted benchmarks; looks pretty similar before and after. This makes sense to me since `get` is not called very often compared to `aggregate`.